.github/workflows/build.yml

```diff
@@ -0,0 +1,66 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+name: Build
+jobs:
+  build-amd64:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set the TAG value
+      id: get-TAG
+      run: |
+        echo "$(make -s log | grep TAG)" >> "$GITHUB_ENV"
+    - name: Build container image
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        push: false
+        tags: rancher/hardened-coredns:${{ env.TAG }}-amd64
+        file: Dockerfile
+
+    - name: Run Trivy vulnerability scanner
+      uses: aquasecurity/trivy-action@0.18.0
+      with:
+        image-ref: rancher/hardened-coredns:${{ env.TAG }}-amd64
+        ignore-unfixed: true
+        vuln-type: 'os,library'
+        severity: 'CRITICAL,HIGH'
+        format: 'sarif'
+        output: 'trivy-results.sarif'
+    - name: Upload Trivy scan results to GitHub Security tab
+      uses: github/codeql-action/upload-sarif@v2
+      if: always()
+      with:
+        sarif_file: 'trivy-results.sarif'
+
+  build-arm64:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Set the TAG value
+      id: get-TAG
+      run: |
+        echo "$(make -s log | grep TAG)" >> "$GITHUB_ENV"
+    - name: Build container image
+      uses: docker/build-push-action@v5
+      with:
```

Comment:
Any specific reason to break the builds per platform? One of the key benefits of docker buildx is [optimised cross-compilation builds](https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/).

Reply:
@pjbgf I think it might be because of this issue: https://github.com/docker/buildx/issues/59

> The default image store in Docker Engine doesn't support loading multi-platform images. You can enable the containerd image store, or push multi-platform images is to directly push to a registry

Reply:
If we are solely testing the build (e.g. confirm whether it works for all target platforms), you can remove both `--load` and `--push` and set `--platform`. That effectively means you will build to all target platforms. Like we did [here](https://github.com/rancher/cis-operator/blob/master/Makefile#L35-L38).